internal/elasticsearch/security/role_mapping.go

```diff
@@ -0,0 +1,240 @@
+package security
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"strings"
+)
+
+func ResourceRoleMapping() *schema.Resource {
+	roleMappingSchema := map[string]*schema.Schema{
+		"id": {
+			Description: "Internal identifier of the resource",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"name": {
+			Description: "The name of the role mapping.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+		},
+		"enabled": {
+			Description: "Specifies whether the role mapping is enabled.",
+			Type:        schema.TypeBool,
+			Required:    true,
+		},
+		"roles": {
+			Description: " A list of role names that are granted to the users that match the role mapping rule. Default is [].",
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Computed:    true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"role_templates": {
```

Comment:
It would be better to define this `role_templates` as a `Set` of `map[string]*schema.Schema` to make sure it can be defined multiple times and use the proper TF domain specific language instead of just providing `json`

Reply:
Sure thing, will try to add tomorrow/day after.